azure/aks/cluster.go

```diff
@@ -0,0 +1,96 @@
+package aks
+
+import (
+	"encoding/base64"
+
+	"github.com/DataDog/test-infra-definitions/azure"
+	"github.com/DataDog/test-infra-definitions/common/utils"
+	"github.com/pulumi/pulumi-azure-native-sdk/containerservice"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+)
+
+const (
+	adminUsername = "azureuser"
+)
+
+func NewCluster(e azure.Environment, name string, nodePool containerservice.ManagedClusterAgentPoolProfileArray, opts ...pulumi.ResourceOption) (*containerservice.ManagedCluster, pulumi.StringOutput, error) {
+	sshPublicKey, err := utils.GetSSHPublicKey(e.DefaultPublicKeyPath())
+	if err != nil {
+		return nil, pulumi.StringOutput{}, err
+	}
+
+	// Warning: we're modifying passed array as it should normally never be used anywhere else
+	nodePool = append(nodePool, systemNodePool(e, "system"))
+
+	cluster, err := containerservice.NewManagedCluster(e.Ctx, e.Namer.ResourceName(name), &containerservice.ManagedClusterArgs{
+		ResourceName:      e.CommonNamer.DisplayName(pulumi.String(name)),
+		ResourceGroupName: pulumi.String(e.DefaultResourceGroup()),
+		KubernetesVersion: pulumi.String(e.KubernetesVersion()),
+		AgentPoolProfiles: nodePool,
+		LinuxProfile: &containerservice.ContainerServiceLinuxProfileArgs{
+			AdminUsername: pulumi.String(adminUsername),
+			Ssh: containerservice.ContainerServiceSshConfigurationArgs{
+				PublicKeys: containerservice.ContainerServiceSshPublicKeyArray{
+					containerservice.ContainerServiceSshPublicKeyArgs{
+						KeyData: sshPublicKey,
+					},
+				},
+			},
+		},
+		AutoUpgradeProfile: containerservice.ManagedClusterAutoUpgradeProfileArgs{
+			// Disabling upgrading as this a temporary cluster, we don't want any change after creation
+			UpgradeChannel: pulumi.String(containerservice.UpgradeChannelNone),
+		},
+		DnsPrefix: pulumi.Sprintf("%s-dns", name),
+		ApiServerAccessProfile: containerservice.ManagedClusterAPIServerAccessProfileArgs{
+			EnablePrivateCluster: pulumi.BoolPtr(false),
+		},
+		NetworkProfile: containerservice.ContainerServiceNetworkProfileArgs{
+			NetworkPlugin: pulumi.String(containerservice.NetworkPluginKubenet),
+		},
+		Identity: containerservice.ManagedClusterIdentityArgs{
+			Type: containerservice.ResourceIdentityTypeSystemAssigned,
+		},
+		Tags: e.ResourcesTags(),
+	}, pulumi.Provider(e.Provider))
+	if err != nil {
+		return nil, pulumi.StringOutput{}, err
+	}
+
+	creds := containerservice.ListManagedClusterUserCredentialsOutput(e.Ctx,
+		containerservice.ListManagedClusterUserCredentialsOutputArgs{
+			ResourceGroupName: pulumi.String(e.DefaultResourceGroup()),
+			ResourceName:      cluster.Name,
+		}, pulumi.Provider(e.Provider),
+	)
+
+	kubeconfig := creds.Kubeconfigs().Index(pulumi.Int(0)).Value().
```

Comment:
Can `creds.Kubeconfigs()` be empty? Is there an explicit error message?

Reply:
It should not be but `ListManagedClusterUserCredentialsOutput` does not return any error. I believe there's an error check inside the `ApplyT` that should make it fail.

Normally it should not be empty.